numpy/core/src/multiarray/compiled_base.c

```diff
@@ -1112,6 +1112,7 @@ arr_ravel_multi_index(PyObject *self, PyObject *args, PyObject *kwds)
         Py_XDECREF(op[i]);
     }
     npy_free_cache_dim_obj(dimensions);
+    /* no NpyIter_Close required since output is always allocated */
```

Comment:
Is there any reason not to just always call it? Testing a handful of flags should be pretty cheap, and if its not, you could cache them in the constructor?

Reply:
We store the need to resolve the writeback in a per-operand flag, so traversing those would be very cheap.

This PR is more to check if our testing is adequate, if there were places that are missing a needed `NpyIter_Close` that would indicate missing tests. It seems that part is OK, barring mistakes.